### PR TITLE
Deterministic retry endpoints

### PR DESCRIPTION
**Pull request by jcorbin**
This disentagles the mocked random from endpoint disposition.  Otherwise it's just too hard to reason about what happens in upcoming changes... plus this seems simpler and more correct anyhow (obvious in retrospect).

r @kriskowal @Raynos

### CODE REVIEW

**Comment by Raynos** [comment]
lgtm.

File: node/test/retry.js [hunk 1]
```diff
@@ -27,42 +27,34 @@ var TChannel = require('../channel');
...
+        .register('foo', servedByFoo(4));
 
     var client = TChannel({
         timeoutFuzz: 0,
-        random: random
+        random: randSeq([
+            1.0, 0.1, 0.1, 0.1, // .request, chan 1 wins
+                 1.0, 0.1, 0.1, // .request, chan 2 wins (1 is skipped)
+                      1.0, 0.1, // .request, chan 3 wins (1-2 are skipped)
+                           0.1  // .request, chan 4 wins (only one left)
```
Comment (kriskowal) [inline node/test/retry.js]: Dyslexics will have an easier time of this if you change 1.0 to 2.0, making it easier to distinguish from 0.1.

**Comment by kriskowal** [comment]
Interesting technique :ok:

File: node/test/retry.js (hunk 1 above)
Comment (jcorbin) [inline node/test/retry.js]: 2.0 would be an invalid value for a random function to return, hence the constant

**Pull request merged**